src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -41,6 +43,13 @@ public function validate(mixed $value, Constraint $constraint)
         $collectionElements = [];
         $normalizer = $this->getNormalizer($constraint);
         foreach ($value as $element) {
+            if (!empty($fields)) {
```

Comment:
we try to not use empty() as much as possible, i'd suggest rewriting this as (grouping the two ifs):
`if ($fields && !$element = $this->reduceElementKeys($fields, $element)) {`

Reply:
Nice, fixed.